### PR TITLE
Usage of a published version of reactor-ts

### PR DESCRIPTION
**Pull request by petervdonovan**
Fixes https://github.com/lf-lang/lingua-franca/issues/1359
Tasks completed:
- [x] Remove `reactor-ts` submodule
- [x] Facilitate local testing using a command line `./gradlew test --tests org.lflang.tests.runtime.TypeScriptTest.* -Druntime="file:<some-path-to-a-local-reactor-ts>"`
- [x] Setup `reactor-ts` CI to run integration tests against a particular commit
- [x] Provide documentation about new development workflow

### CODE REVIEW

File: org.lflang/src/lib/ts/tsconfig.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
         "strictBindCallApply": true,
         "strictNullChecks": true,
         "strictFunctionTypes": true,
-        "typeRoots": ["./node_modules/@types/", "./node_modules/reactor-ts/src/core/@types/"]
+        "typeRoots": ["./node_modules/@types/", "./node_modules/@lf-lang/reactor-ts/src/core/@types/"]
```
Comment (lhstrh) [inline org.lflang/src/lib/ts/tsconfig.json:16]: Note: the second element of the list should no longer be necessary once we remove our dependency on `ulog` as I suggested.

File: org.lflang.tests/src/org/lflang/tests/TestBase.java [hunk 2]
```diff
@@ -384,8 +384,21 @@ private static void checkAndReportFailures(Set<LFTest> tests) {
      * @throws IOException if there is any file access problem
      */
     private LFGeneratorContext configure(LFTest test, Configurator configurator, TestLevel level) throws IOException {
+        var props = new Properties();
+        var sysProps = System.getProperties();
+        // Set the external-runtime-path property if it was specified.
+        if (sysProps.containsKey("runtime")) {
```
Comment (lhstrh) [inline org.lflang.tests/src/org/lflang/tests/TestBase.java:390]: Note to self: this code is outdated and won't work.

File: org.lflang.cli/src/org/lflang/cli/Lfc.java [hunk 3]
```diff
@@ -95,20 +96,18 @@ enum CLIOption {
         /**
          * Construct a new CLIOption.
          *
-         * @param opt         The short option name. E.g.: "f" denotes a flag
-         *                    "-f".
-         * @param longOpt     The long option name. E.g.: "foo" denotes a flag
-         *                    "--foo".
-         * @param hasArg      Whether or not this option has an argument. E.g.:
+         * @param parameter   teThe build parameter that this CLI parameter corresponds to.
```
Comment (petervdonovan) [inline org.lflang.cli/src/org/lflang/cli/Lfc.java:99]:
```suggestion
         * @param parameter   The build parameter that this CLI parameter corresponds to.
```

**Review by petervdonovan — COMMENTED** [review]
I can't approve this because of how GitHub behaves, but it looks right to me. I agree that it is cleaner to introduce the `BuildParm` enum.

File: test/TypeScript/README.md [hunk 4]
```diff
@@ -1,2 +1,19 @@
...
+```
+./gradlew test --tests org.lflang.tests.runtime.TypeScriptTest.* -Druntime="~/lf-lang/reactor-ts"
+```
+- To point to a particular ref (e.g., `feature-branch`) of the `reactor-ts` repository:
+```
+./gradlew test --tests org.lflang.tests.runtime.TypeScriptTest.* -Druntime="git://github.com/lf-lang/reactor-ts.git#feature-branch"
+```
+- Note that `lfc` can be pointed to an alternative runtime as well, using the `external-runtime-path` switch:
+```
+lfc test/TypeScript/src/Minimal.lf --external-runtime-path ~/lf-lang/reactor-ts
```
Comment (cmnrd) [inline test/TypeScript/README.md:18]: Note that for lfc we also have the `--runtime-version` argument which is intended for specifying a particular ref to be fetched from the main runtime repository. Is this also supported for TS now? If not, I think we should add it.

**Review by cmnrd — APPROVED** [review]
This looks good to me! The improvements for Linux and Windows are significant, but oddly on MacOS the tests are still running after 2 hours. Please also consider my comment regarding `--runtime-version`

File: test/TypeScript/README.md (hunk 4 above)
Comment (lhstrh) [reply test/TypeScript/README.md:14]: OK, I've added this (and removed the overloading of the `-Druntime` switch as it 1) isn't strictly necessary and 2) doesn't seem to allow `pnpm` to reuse prior installations of `@lf-lang/reactor-ts`. Instead, in CI, I'm just checking out the repo and pointing to the location in the filesystem for tests invoked as part of `lf-lang/reactor-ts` CI runs.

**Review by lhstrh — APPROVED** [review]

**Pull request merged**